src/Gitlab/Client.ts

```diff
@@ -71,6 +72,29 @@ export class GitLabClient {
         }
     }
 
+    private async listProjects(minAccess: AccessLevel, inGroup?: ProjectId, idAfter?: number, search?: string): Promise<SimpleProject[]> {
+        try {
+            const path = inGroup ? `api/v4/groups/${getProjectId(inGroup)}/projects` : 'api/v4/projects';
```

Comment:
What is `inGroup`? The type `ProjectId` does not seem to make sense.

Reply:
`ProjectId`  was a messy way to support path segments (string[]), paths (string) and ids (number). It's a total mess though so I've cleaned it up.

This specific line is changing the API call depending on whether we are searching within a group for some projects, or searching globally.